File: javascript/hamming/hamming.js
var R = require('ramda');

var Hamming = function(){
  function compute(strand1, strand2) {
    if (strand1.length !== strand2.length) throw Error("DNA strands must be of equal length.")
    return R.zip(strand1, strand2).filter(([a, b]) => a !== b).length
  }

  return {
    compute: compute
  }
}

module.exports = Hamming

File: java/hamming/src/main/java/Hamming.java
class Hamming {

    String leftStrand;
    String rightStrand;
    Hamming(String leftStrand, String rightStrand) {
        if (leftStrand.length() != rightStrand.length()) {
            throw new IllegalArgumentException("leftStrand and rightStrand must be of equal length.");
        }

        this.leftStrand = leftStrand;
        this.rightStrand = rightStrand;
    }

    int getHammingDistance() {
        int counter = 0;
        int maxDifference = 0;
        for (int i=0; i<leftStrand.length(); i++) {
            if (leftStrand.charAt(i) != rightStrand.charAt(i)) {
                counter++;
            }
            if (counter > maxDifference) {
                maxDifference = counter;
            }
        }

        return maxDifference;
    }
}
